server/index.js

```diff
@@ -154,10 +154,47 @@ app.patch('/api/noshare/:entryId', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.patch('/api/sharedit/:entryId', (req, res, next) => {
+  const entryId = Number(req.params.entryId);
+  if (!entryId) {
+    throw new ClientError(400, 'entryId must be a positive integer');
+  }
+  const sql = `
+  update "code-journal" set "sharedEdit"='true' where "entryId"=$1 returning *
+  `;
+  const codeArray = [entryId];
+  db.query(sql, codeArray)
+    .then(result => {
+      if (!result.rows[0]) {
+        throw new ClientError(404, `cannot find entry with entryId ${entryId}`);
+      }
+      res.json(result.rows[0]);
+    })
+    .catch(err => next(err));
+});
+app.patch('/api/nosharedit/:entryId', (req, res, next) => {
+  const entryId = Number(req.params.entryId);
+  if (!entryId) {
+    throw new ClientError(400, 'entryId must be a positive integer');
+  }
+  const sql = `
+  update "code-journal" set "sharedEdit"='false' where "entryId"=$1 returning *
+  `;
+  const codeArray = [entryId];
+  db.query(sql, codeArray)
+    .then(result => {
+      if (!result.rows[0]) {
+        throw new ClientError(404, `cannot find entry with entryId ${entryId}`);
+      }
+      res.json(result.rows[0]);
+    })
+    .catch(err => next(err));
```

Comment:
Ok, these two routes are almost the same. try to make it one route only. also, you should only allow the owner to make it shareEdit true